test/controllers/wizards_controller_test.rb

```diff
@@ -1,5 +1,68 @@
 require "test_helper"
-
+require 'pry'
 describe WizardsController do
   
+  
+  
+  describe "auth_callback" do
+    it "logs in an existing wizard and redirects to the root route" do
+      start_count = Wizard.count
+      wizard = wizards(:wizard1)
+      
+      perform_login(wizard)
+      
+      must_redirect_to root_path
+      expect(session[:wizard_id]).must_equal wizard.id
+      Wizard.count.must_equal start_count
+    end
+    
+    it "creates an account for a new user and redirects to the root route" do
+      new_wizard = Wizard.new(provider: "github", uid: 99999, username: "testing", email: "test@wizard.com")
+      # binding.pry
+      expect{
```

Comment:
my code has a bunch of comments I need to delete too I think we can all scan through everything later